projectorganizer/src/prjorg-sidebar.c

```diff
@@ -300,6 +300,181 @@ static void on_follow_active(GtkToggleToolButton *button, G_GNUC_UNUSED gpointer
+			tree_path = gtk_tree_model_get_path (model, &iter);
+
+			utf8_path = build_path(&iter);
+			locale_path = utils_get_locale_from_utf8(utf8_path);
+			g_free(utf8_path);
+		}
+		else
+		{
+			GIcon *icon;
+			gtk_tree_model_get(model, &iter, FILEVIEW_COLUMN_ICON, &icon, -1);
+			if (icon)
+			{
+				gchar *utf8_path;
+				utf8_path = build_path(&iter);
+				locale_path = utils_get_locale_from_utf8(utf8_path);
+
+				g_free(utf8_path);
+				g_object_unref(icon);
+			}
+		}
+	}
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_EXISTS))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	/* try to get path from document */
+	if (!locale_path)
+	{
+		GeanyDocument *doc = NULL;
+		doc = document_get_current();
+		if (DOC_VALID(doc))
+			locale_path = g_strdup(doc->real_path);
+	}
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_EXISTS))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	/* try to get path from project */
+	if (!locale_path && geany_data->app->project)
+	{
+		gchar *utf8_path = NULL;
+		utf8_path = get_project_base_path();
+		locale_path = utils_get_locale_from_utf8(utf8_path);
+		g_free(utf8_path);
+	}
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_EXISTS))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	/* Get parent folder if regular file */
+	if (locale_path && g_file_test(locale_path, G_FILE_TEST_IS_REGULAR))
+	{
+		gchar *folder;
+		folder = g_path_get_dirname(locale_path);
+		if (folder && folder == '.')
+		{
+			g_free(folder);
+			folder = NULL;
+		}
+		if (folder && g_file_test(folder, G_FILE_TEST_IS_DIR))
+		{
+			g_free(locale_path);
+			locale_path = folder;
+		}
+		else
+		{
+			g_free(folder);
+			g_free(locale_path);
+			locale_path = NULL;
+		}
+	}
+
+	/* Try home folder */
+	if (!locale_path)
+		locale_path = g_strdup(g_get_home_dir());
+
+	/* Make sure path exists */
+	if(locale_path && !g_file_test(locale_path, G_FILE_TEST_IS_DIR))
+	{
+		g_free(locale_path);
+		locale_path = NULL;
+	}
+
+	return locale_path;
+}
+
+
+void on_open_file_manager(G_GNUC_UNUSED GtkMenuItem * menuitem, G_GNUC_UNUSED gpointer user_data)
+{
+	gchar *locale_path;
+	gchar *open_command;
+
+	open_command = PRJORG_COMMAND_OPEN;
+	locale_path = get_folder_for_selection();
+
+	if (locale_path)
+	{
+		gchar *command;
+		command = g_strconcat (open_command, " \"", locale_path, "\"", NULL);
+		if (!spawn_async(locale_path, command, NULL, NULL, NULL, NULL))
+			msgwin_status_add(_("Unable to open folder.  Command unavailable: %s"), open_command);
+
+		g_free(command);
+		g_free(locale_path);
+	}
+	else
+	{
+		msgwin_status_add(_("Unable to find folder."));
+	}
+}
+
+
+void on_open_terminal(G_GNUC_UNUSED GtkMenuItem * menuitem, G_GNUC_UNUSED gpointer user_data)
+{
+	gchar *locale_path, *open_command;
+
+	if(g_file_test(PRJORG_COMMAND_TERMINAL_ALT, G_FILE_TEST_EXISTS))
```

Comment:
Space after 'if'.

Reply:
Thanks.  I'm used to just letting `clang-format` clean up code.